### DISCUSSION TITLE
Need Help on the Blockchain creation using NodeJS

### DISCUSSION THREAD

**Discussion by RajeshkumarManoharan** [ac12644/Blockchain#3] [General]
Hi All,

I am able to start the p2p network from both the terminal in mac os with this Blockchain code but both terminals are not connected to each other as Peers.
Kindly help to understand if anything missing in code or environment configuration.

Also advise how to test the adding new block and register miners.

Thanks & Regards,
M Rajeshkumar

**Comment by damartripamungkas** [comment]
that's because your swarm is still in the local computer environment, so you have to expose your swarm to the public but I don't understand how

**Comment by ac12644** [comment]
for this, you have to deploy the app on a cloud network. For AWS you can follow these steps:

1. Launch a new EC2 instance from the AWS Management Console.
2. Set the security group settings to open necessary ports. You have to predefine the port in p2p.js and need to open the HTTP (<PORT NUMBER>), HTTPS (<PORT NUMBER>), or any other port your application uses.
3. SSH into the created instance using the provided .pem file.
4. Install Node.js on the instance.
5. Upload your application to the instance using git clone.
6. Run the application on the instance with npm install and node.
7. Access your application using the public IP address or DNS of the instance.
